Prototype for moving List procedures page to Ops Manual (#3003)
neo4j/docs-operations: modules/ROOT/pages/procedures/show-procedures.adoc

:description: Information about the `SHOW PROCEDURES` command.
:test-skip: true
= Show procedures

You can use `SHOW PROCEDURES` to list all available procedures.
For general information about the `SHOW` command, see the link:{neo4j-docs-base-uri}/cypher-manual/clauses/show/[Cypher Manual -> SHOW].

[[syntax]]
== Syntax

For full details about the syntax descriptions, see xref:database-administration/syntax.adoc#administration-syntax-reading[Administration command syntax].

[options="header", width="100%", cols="1,5a"]
|===
| Action | Syntax

| Show all procedures
|
[source, syntax, role="noheader"]
----
SHOW PROCEDURE[S]
[YIELD { * \| field[, ...] } [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
[WHERE expression]
[RETURN field[, ...] [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
----

> | Show procedures executable by the current user

Reviewer: Previously we had this (probably in an attempt to give a slightly different description than what the clause itself does), do we want to still do something like that or are we fine with repeating the wording of the clause?
```suggestion
| Show procedures that the current user can execute
```

|
[source, syntax, role="noheader"]
----
SHOW PROCEDURE[S] EXECUTABLE [BY CURRENT USER]
[YIELD { * \| field[, ...] } [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
[WHERE expression]
[RETURN field[, ...] [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
----

> a| Show procedures executable by a specified user

Reviewer: same here
```suggestion
a| Show procedures that the specified user can execute
```

| LIST<MAP>
|

| admin
| `true` if this procedure is an admin procedure.
| BOOLEAN
|

| rolesExecution
| List of roles permitted to execute this procedure.
Is `null` without the link:https://neo4j.com/docs/operations-manual/current/authentication-authorization/dbms-administration/dbms-role-management-privileges/#_grant_privilege_to_show_roles[`SHOW ROLE`] privilege.
| LIST<STRING>
|

| rolesBoostedExecution
| List of roles permitted to use boosted mode when executing this procedure.
Is `null` without the link:https://neo4j.com/docs/operations-manual/current/authentication-authorization/dbms-administration/dbms-role-management-privileges/#_grant_privilege_to_show_roles[`SHOW ROLE`] privilege.
| LIST<STRING>
|

| isDeprecated
| Whether the procedure is deprecated.
| BOOLEAN
|

| deprecatedBy
| The replacement procedure to use in case of deprecation; otherwise `null`.
| STRING
|

| option
| Map of extra output, e.g. if the procedure is deprecated.
| MAP
|
|===

The deprecation information for procedures is returned both in the `isDeprecated` and `option` columns.

[[examples]]
== Examples

.Show all procedures and return only default columns

> [source, cypher, role=test-result-skip]

Reviewer: you put `:test-skip: true` at the top of the page, do you also need `role=test-result-skip` on each example 🤔

Author: We don't need any of them.

----
SHOW PROCEDURES
----

.Show all procedures and return all columns
[source, cypher, role=test-result-skip]
----
SHOW PROCEDURES YIELD *
----

.Show all procedures and return specific columns only
[source, cypher, role=test-result-skip]
----
SHOW PROCEDURES YIELD name, admin
----

.Filter `SHOW PROCEDURES` using `WHERE`
[source, cypher]
----
SHOW PROCEDURES YIELD name, worksOnSystem

> WHERE worksOnSystem = TRUE

Reviewer: worksOnSystem is already a boolean, we don't need to check that `TRUE = TRUE` :P
```suggestion
WHERE worksOnSystem
```

Author: True, but is this version of the query not more readable (user's may not know it is a boolean)?

Reviewer: sure, don't really feel that strongly about it (though we do list the return columns and their type higher up on the page if the fact that we only used the variable wasn't enough to clue you in to it being a boolean :P)

RETURN name
----

> .Show procedures executable by the current user

Reviewer: I feel like we're missing this information from the old page
> The listed procedures can also be filtered by whether a user can execute them.
This filtering is only available through the `EXECUTABLE` clause and not through the `WHERE` clause.
This is due to using the user's privileges instead of filtering on the available output columns.

Specifically the fact that EXECUTABLE doesn't filter based on the output but based on the privileges of the relevant user (current or specified)

[source, cypher, role=test-result-skip]
----
SHOW PROCEDURES EXECUTABLE BY CURRENT USER YIELD *
----

> The second option for using the `EXECUTABLE` clause is to filter the list to only contain procedures executable by a specific user.

Reviewer: Feels odd to say `second option` when we don't have any sentence about `first option` anymore
> There are two options for using the `EXECUTABLE` clause.
The first option is to filter for the current user:

The following example shows the procedures available to the user `jake`, who has been granted the `EXECUTE PROCEDURE dbms.*` privilege by the `admin` of the database.
For more information about `DBMS EXECUTE` privilege administration, see the xref:authentication-authorization/dbms-administration/dbms-execute-privileges.adoc[The `DBMS EXECUTE` privileges].

.Show procedures executable by specific user
[source, cypher, role=test-result-skip]
----
SHOW PROCEDURES EXECUTABLE BY jake
----

[role=label--new-2026.05 label--cypher-25]
[[composable-show-procedures]]

> == Combine `SHOW PROCEDURES` with other Cypher commands

Reviewer: I don't think we usually uses `command` for the different parts of Cypher other than the show, index, schema and administration commands. Otherwise we just use `clause`

```suggestion
== Combine `SHOW PROCEDURES` with other Cypher clauses
```

I guess this might be more general and maybe not limited to only the show procedures page, but I noticed it here.

> `SHOW PROCEDURES` can be combined with other Cypher commands to form a single query.

Reviewer: ```suggestion
`SHOW PROCEDURES` can be combined with other Cypher clauses to form a single query.
```
